crates/ide-completion/src/tests/type_pos.rs

```diff
@@ -719,3 +700,264 @@ pub struct S;
         "#]],
     )
 }
+
+#[test]
+fn completes_const_and_type_generics_separately() {
+    // Function generic params
+    check(
+        r#"
+    struct Foo;
+    const X: usize = 0;
+    fn foo<T, const N: usize>() {}
+    fn main() {
+        foo::<F$0, _>();
+    }
+            "#,
+        expect![[r#"
+                en Enum
+                ma makro!(…) macro_rules! makro
+                md module
+                st Foo
+                st Record
+                st Tuple
+                st Unit
+                tt Trait
+                un Union
+                bt u32
+                kw crate::
+                kw self::
+            "#]],
+    );
+    check(
+        r#"
+    struct Foo;
+    const X: usize = 0;
+    fn foo<T, const N: usize>() {}
+    fn main() {
+        foo::<_, $0>();
+    }
+            "#,
+        expect![[r#"
+                ct CONST
+                ct X
+                ma makro!(…) macro_rules! makro
+                kw crate::
+                kw self::
+            "#]],
+    );
```

Comment:
Here we actually still want to show the `Foo` struct because it could potentially have an associated const taht we want to complete. Its fine to put a FIXME on this for now as to not overload this PR with too many things.

Reply:
Added a FIXME. Also not sure how this should interact with the fact that `foo::<usize::MAX>()` is invalid (it should be `foo::<{ usize::MAX }>()`). Would be nice if we could get the completion to suggest `{ usize::MAX }` instead of just `usize`, which makes less sense if you don't already know it has associated constants

Reply:
> Could we complete associated constants e.g. `Foo::X` if they exist but not `Foo` itself?

We probably could, likewise for const function calls